src/Next.php

```diff
@@ -39,12 +41,15 @@ public function __construct(SplQueue $queue, RequestHandlerInterface $fallbackHa
     {
         $this->queue           = clone $queue;
         $this->fallbackHandler = $fallbackHandler;
+        $this->queue->push(
+            new RequestHandlerMiddleware($fallbackHandler)
+        );
     }
 
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
         if ($this->queue->isEmpty()) {
```

Comment:
Reuse queue as a flag where null means there is no queue to process, unlike empty queue where fallback handler goes into play
```suggestion
         if ($this->queue === null) {
             throw MiddlewarePipeNextHandlerAlreadyCalledException::create();
         }
         if ($this->queue->isEmpty()) {
```